### DISCUSSION TITLE
Integration: viem-agglayer

### DISCUSSION THREAD

**Discussion by pgebheim** [agglayer/agglayer#75] [Ideas]
Dapp and frontend devs are going to need to work with AggLayer. Let's integrate with the tools they're already using! 

A viem integration should exist with helpers and utilities around L2-L2 interactions, with a similar design pattern that is used for the viem-op extension.

Feature Ideas:
- extend clients with aggregated functionality
- bridge and call support
- isAggregated helper to detect if your network is connected to the shared bridge
- message claim helpers to make it easy for Dapp devs to allow users to perform message claims

**Comment by pgebheim** [comment]
By the way existing resources here:
- https://github.com/0xpolygon/lxly.js
- https://github.com/AggLayer/lxly-bridge-and-call
- https://github.com/0xrouss/agglayer-scripts

**Comment by 0xrouss** [comment]
Hi, I started it [agglayer-viem](https://github.com/0xrouss/agglayer-viem)

I will be adding more features, if you have a list with the features needed let me know.

When it achieves a good version we can try to add it directly to viem

**Reply by pgebheim** [reply]
Awesome. 

Perhaps the main goal should be to support the features in LxLy.js in a viem way:

- fetch claim proofs from bridge service
- do claim txns on destination chain
- integrated bridge and call

I also think we should separately make and agglist, similar to token list and which keeps a simple json registry for chains on the AL
